test/test_datapipe.py

```diff
@@ -497,6 +563,56 @@ def _non_bool_fn(data):
         with self.assertRaises(ValueError):
             temp = list(d for d in filter_dp)
 
+    def test_filter_datapipe_nested_list(self):
+
+        input_ds = IDP(range(10)).batch(5)
+
+        def _filter_fn(data, val):
+            return data >= val
+
+        filter_dp = input_ds.filter(nesting_level=-1, filter_fn=_filter_fn, fn_kwargs={'val': 5})
+        expected_dp1 = [[5, 6, 7, 8, 9]]
+        self.assertEqual(len(list(zip(filter_dp, expected_dp1))), len(expected_dp1))
```

Comment:
IMO, `self.assertEqual(len(list(d for d in filter_d)), len(expected_dp1))` is better to verify the length remains same. If you use `zip` and `filter_dp` has longer length somehow, the verification still pass unexpectedly, because `zip` would iterate based on the smaller length.